src/App.js

```diff
@@ -1,24 +1,116 @@
+  getCurrentCountryIndex,
+} from "./features/country/countrySlice";
+
+import { openOptions } from "./features/list/listSlice";
 
 function App() {
+  const {
+    isLoading,
+    error,
+    countriesInfo,
+    countryName,
+    loadingMoreCountries,
+    isInitialSearch,
+    filteredCountry,
+    currentCountryIndexToNumber,
+  } = useSelector((state) => state.country);
+  const { isOpenOptions } = useSelector((state) => state.list);
+  const dispatch = useDispatch();
+
+  const checkCountryExisting = countriesInfo.map(
+    (country) => country.name === countryName
+  );
+
+  const handleClick = (e) => {
+    e.target.placeholder = "Search";
+    dispatch(openOptions());
+    dispatch(initialSearch());
+    dispatch(getFilteredCountry(countryName));
+    dispatch(getCurrentCountryIndex(countryName));
+
+    const regex = /[\uD83C][\uDDE6-\uDDFF][\uD83C][\uDDE6-\uDDFF]\s+/;
+    if (regex.test(countryName)) {
+      dispatch(getCountryName(countryName));
+      dispatch(getCurrentCountryIndex(countryName));
+    }
+    if (currentCountryIndexToNumber === 0 && 1 && 248 && 249) {
+      dispatch(notInitialSearch());
+    }
+
+    if (countryName && checkCountryExisting && !undefined) {
+      dispatch(getMoreOptions(filteredCountry));
+      dispatch(notInitialSearch());
+    }
+  };
+
+  const handleSearchCountryName = (e) => {
+    dispatch(getCountryName(e.target.value));
+  };
+
+  useEffect(() => {
+    dispatch(fetchCountries("name"));
+  }, [dispatch]);
+
+  useEffect(() => {
+    dispatch(getFilteredCountry(countryName));
+  }, [countryName, dispatch]);
+
+  if (error) {
+    return (
+      <div className="loading">
+        <h1>Something went wrong </h1>
+      </div>
+    );
+  }
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <>
+      {isLoading ? (
+        <div className="App">
+          <h1>Loading ... </h1>
+        </div>
+      ) : (
+        <div className="App">
+          <input
+            type="text"
+            placeholder="Select"
+            onChange={handleSearchCountryName}
+            onClick={handleClick}
+            className="inputContainer"
+            value={countryName || ""}
+          />
+          <small className="arrowDown">▼</small>
+          {isOpenOptions && (
+            <div className="optionContainer">
+              <ul
+                className="ulOption"
+                style={{ overflow: isInitialSearch ? "scroll" : "none" }}
+              >
+                {loadingMoreCountries && loadingMoreCountries.length > 1
+                  ? loadingMoreCountries?.map((country) => (
```

Comment:
```suggestion
                  ? loadingMoreCountries.map((country) => (
```
 if you already used optional chaining on the variable, no need to apply again.

See the comparison results on the [babel compiler](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=PQKgUABAkASgpgZwK4BsAuECMYTDGFAewEMATASwDsBzAWUICc4BhQpStB8xAOgFtiAB3yhIsRKgwAmHHiJkqdRizYcuiAPz8hQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2&prettier=false&targets=&version=7.18.12&externalPlugins=&assumptions=%7B%7D)